Framework/PythonInterface/core/src/PythonStdoutChannel.cpp

```diff
@@ -14,28 +14,30 @@
 #include <boost/iostreams/stream.hpp>
 
 namespace { // anonymous namespace
+
 class PyStdoutSink {
 public:
   typedef char char_type;
   typedef boost::iostreams::sink_tag category;
 
   std::streamsize write(const char *s, std::streamsize n) {
+    Mantid::PythonInterface::GlobalInterpreterLock gil; // acquire the GIL
     // PySys_WriteStdout truncates to 1000 chars
     static const std::streamsize MAXSIZE = 1000;
 
     std::streamsize written = std::min(n, MAXSIZE);
     PySys_WriteStdout((boost::format("%%.%1%s") % written).str().c_str(), s);
 
     return written;
-  }
+  } // release the GIL
```

Comment:
This seems to be a "fix".

Reply:
It is a fix.
I wrote a new new python channel but removed it later, seeing that acquiring the GIL fixed this channel